test/extended/oauth/requestheaders.go

```diff
@@ -207,6 +207,10 @@ var _ = g.Describe("[Serial] [sig-auth][Feature:OAuthServer] [RequestHeaders] [I
 
 		for _, tc := range testCases {
 			g.By(tc.name, func() {
+				g.AfterEach(func() {
```

Comment:
`AfterEach()` is run after every `It()`. I suppose we could rewrite the test so that this is `It()` instead of `By()`, in which case you should move the `AfterEach()` outside. Also, I think we only want the PostMortem in case `g.CurrentGinkgoTestDescription().Failed` is `true`

Reply:
Ok, then I'll revert the change and do it as @slaskawi did it.

> Documenting Complex Specs: By

`By` is meant to be some kind of complex pre-condition for a test case (`it`). I don't see any `By` preparation with `It` test cases, so I don't want to be adhere more to ginkgo then the  test authors.